### DISCUSSION TITLE
Artisan doesn't read temperature of BK Roasting, stay in zero

### DISCUSSION THREAD

**Discussion by Sacha2306** [artisan-roaster-scope/artisan#875] [Q&A]
Hi everyone, my name is Sacha, i'm from Argentina. I'm working with a Blue King Roaster of 6 kgs connected whit Artisan Scope. I'm having a problem that appeared alone, i didn't change the settings of ports or anything like that. Was like a restart of software and then stopped giving me information like temperature of beans and exhaust. I think that it suffer a actualization. I will let you screen shots of what happening. 
I hope that the community would help me and i will appreciate so much. 

A pleasure!
![config artisan 2](https://user-images.githubusercontent.com/104920123/166724199-035f8b9c-12c9-466c-89e4-33d1d4889d89.jpg)

![config artisan](https://user-images.githubusercontent.com/104920123/166724160-809a03ea-08db-4e0c-8b40-f57082e4147d.jpg)

**Comment by poundy** [comment]
how long have you been using Artisan 2.6.0? Did you do successful roasting since that upgrade?

I have a BK manufactured roaster and I am pretty sure that my settings are not the same as yours! Have you changed them? 
From your second screensot, my timeout is 0.4sec not 2.0. I also don't have little-endian bytes ticked, and i have only one stop-bit not 2 like you do. So, this makes me suspicious... 

Do you have an archived version of your settings that you manually saved some time ago?  If so, can you upload your current setting and your previous setting, making it clear inthe file names what they are? Note you have to rename the files with a .TXT extension to drag and drop them here in github.

**Reply by Sacha2306** [reply]
Thank u for the answer. I been using it just few days ago, and i didn't a successful roasting whit it. 

I just put the settings like u and it didn't response. 

I have only archived my current setting, i'm new in this and this things escape from me hehe. But i tried whit other computer that has the previous settings, the same problem, but i can archive the information tonight. 
I tried whit other wire to, and the same, i'm thinking that maybe the problem belong to the themperature controller. It's an Omron E5CC. 

I'm gonna let the current settings (Whit the setting like u have)

Whit yesterday's setting
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/8631157/artisan-settings.txt)

Whit current settings
[artisan-settings2.txt](https://github.com/artisan-roaster-scope/artisan/files/8631161/artisan-settings2.txt)

Thank you again!

**Reply by poundy** [reply]
ok I will look at your settings compared to mine at some point (I am away at the moment).

So just to be clear. You have never used Artisan successfully, any version, talking to this roaster, on any computer? 

Is this a brand new roaster, or just a roaster that you are newly trying to connect to (so perhaps you have just bought it)?  What I was hoping to get by asking for previous settings, was a known-working configuration that we could then use to move forward to your current config.  If this is not a new roaster, and someone else has used roasting software, perhaps they can elaborate on the correct settings.  What makes you suspect the Omron, is there something you're aware of that says it is faulty or was replaced ?

Fundamentally, these roasters communicate over modbus. If anything in the modbus system has been set wrongly or was changed from defaults, there's a chance nothing will work.  So please if you can, give the history of the machine and how / if it was ever used with roasting software (of any kind).  If it's brand new, then I would suggest you reset Artisan to defaults and then use the wizard to set it up as a BlueKing; that should set it up to what the BK roaster needs, but I've seen manufacturers change hardware settings and not tell anyone (but can't recall seeing this about BK).

**Reply by Sacha2306** [reply]
I have used it successfully, yes. I did more than a hundred roasting whit the 2.4.6 version im thinking, whit this roaster and two diferent computer whit Windows 10. 

I suspect that may be the Omron because i tried whit the second computer that has not the latest version of Artisan and doesn't work to. 

Can you tell me more about the wizard please?

**Reply by poundy** [reply]
ok, so what you are saying is that you've never had success with 2.6.0 on this PC, and you have previously had success on the 2.4.x on another PC and even now it no longer works. Then as long as you are sure you haven't changed Artisan configuration "trying to get it working" then I would say there's a problem with the connection to the roaster or a problem with the components on the roaster.  

Often USB cables are simple failure points; check and replace.  Check and confirm that the serial port is visible on the PC - open Device Manager and check what serial ports are visible once you connect the roaster. Sometimes that port does change (from COM3 to COM5 for example) when no real changes have occurred.

**Reply by poundy** [reply]
The wizard is just going to the MACHINE menu and then selecting your machine and answering those questions.

**Reply by Sacha2306** [reply]
I had never used succesfull the 2.6.0 but the previously version i did, in the same computer. Was like a reboot of the software, beacuse i'm using it in Celcius and it's changed to Farenheit suddenly. 

The serial port i'ts ok, and the USB cable it's ok too. 

About the Wizard, i had set up after the problem, but i'm gonna trie again now

**Reply by bigl00z3** [reply]
When moving to a new version always save you settings in the old software `Help/Save Settings..` and then reimport them in the new version `Help/Load Settings.... `
Check [https://artisan-scope.org/docs/upgrading/](https://artisan-scope.org/docs/upgrading/)

Artisan can work both in °F or °C as required by the user. When you upgraded it likely went to default settings. Most artisan functions and possibilities are well explained in the [QuickStart Guide](https://artisan-scope.org/docs/quick-start-guide/) you should check it, it's very informative.

**Reply by poundy** [reply]
Yes in my mind it seems like you have lost your actual settings and you should either find your original settings file (or a backup as Nico suggests) or go back through the whole setup and re-establish all settings.

**Reply by Sacha2306** [reply]
Guys, sorry. I was out of town. 
I'm thinking that the problem belong to the roast machine. I'm gonna call an a tecnician and ask for take a look. 
I made a lot of movements on the computer and it's look working normally. Maybe the problem is that the machine doesn't share succesfull the information, because the connection on software it's fine, just doesn't give me the temperature that the machine shows me. 
I appreciate a lot your help, really, i hope to on the future i could help you in anything. If you want, we can close this, or we can wait to see if i get to repair it and let the information here for other people.
